doc/2/guides/develop-on-kuzzle/2-api-controllers/index.md

```diff
@@ -8,3 +8,452 @@ order: 200
+        // and the anonymous user ID is "-1"
+        assert(request.context.user._id === '-1')
+        assert(request.context.connection.protocol === 'http')
+      },
+    }
+  }
+})
+```
+
+::: info
+More informations about the [RequestContext](/core/2/some-link) class properties.
+:::
+
+## Response format
+
+Kuzzle Response are **standardized**. This format is shared by all API actions, including custom controller actions.
+
+A Kuzzle Response is a **JSON object** with the following format:
+
+| Property     | Description                                                                                         |
+| ------------ | --------------------------------------------------------------------------------------------------- |
+| `action`     | API action                                                                          |
+| `collection` | Collection name, or `null` if no collection was involved                                       |
+| `controller` | API controller                                                                             |
+| `error`      | [KuzzleError](/core/2/api/essentials/error-handling) object, or `null` if there was no error                |
+| `index`      | Index name, or `null` if no index was involved                                                 |
+| `requestId`  | Request unique identifier                                                                           |
+| `result`     | Action result, or `null` if an error occured                                                         |
+| `status`     | Response status, using [HTTP status codes](https://en.wikipedia.org/wiki/List_of_HTTP_status_codes) |
+| `volatile`   | Arbitrary data repeated from the initial request                                                    |
+The `result` property will contain the return of the action `handler` function.
+
+For example, when calling this controller action:
+```js
+app.controller.register('greeting', {
+  actions: {
+    sayHello: {
+      handler: async (request: Request) => {
+        return `Hello, ${request.input.args.name}`
+      }
+    }
+  }
+})
+```
+
+The following response will be sent:
+
+```bash
+$ npx wscat -c ws://localhost:7512 --execute '{
+  "controller": "greeting",
+  "action": "sayHello",
+  "name": "Yagmur"
+}'
+
+{
+  "requestId": "a6f4f5b6-1aa2-4cf9-9724-12b12575c047",
+  "status": 200,
+  "error": null,
+  "controller": "greeting",
+  "action": "sayHello",
+  "collection": null,
+  "index": null,
+  "volatile": null,
+  "result": "Hello, Yagmur", # <= handler function return value
+  "room": "a6f4f5b6-1aa2-4cf9-9724-12b12575c047"
+}
+```
+
+#### Return a custom response
+
+In some cases it may be necessary to **return a response that differs** from the standard API response format.
+
+This may be to send a **smaller JSON response** for constrained environments, to **perform HTTP redirection** or to **return another mime** type such as CSV, an image, a PDF document, etc.
```

Comment:
```suggestion
This may be to send a **smaller JSON response** for constrained environments, to **perform HTTP redirection** or to **return another MIME type** such as CSV, an image, a PDF document, etc.
```